### PR TITLE
🛠️ DAT-17916: Docker build and push 

### PR DESCRIPTION
**Pull request by sayaliM0412**
1. fix: `.github/dependabot.yml` : add docker `package-ecosystem` so that any new Docker release for OSS is triggered that will be triggered `hourly`. It takes sometime for our PR to be merged to official docker images.
2. feat: `.github/workflows/deploy-extension-to-marketplace.yml` : Docker Build and Push to AWS Marketplace:  triggers the workflow automatically when a pull request is closed (with types: - closed) on the main branch. The workflow only runs if the change involves modifications to the Dockerfile (specified in paths). Also, includes manual triggers.
3. feat: `.github/workflows/extension-update-pom.yml` : automatic update of the pom.xml based on new OSS versions
4. feat: `Dockerfile` : updates the Liquibase Package Manager (LPM) and installs the AWS License Service Extension and sets the default command to show the Liquibase version when the container is run.
5. feat: `README.md`: update readme

### CODE REVIEW

File: .github/workflows/deploy-extension-to-marketplace.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    branches:
+      - main
+    types:
+      - closed
+    paths:
+      - 'Dockerfile'
+
+jobs:
+  build:
+      if: github.actor == 'dependabot' || github.event_name == 'workflow_dispatch'
```
Comment (filipelautert) [inline .github/workflows/deploy-extension-to-marketplace.yml:20]: But having dependabot here also means that everytime we merge a Java dependency we will release it to docker, right? Maybe we need additional checks?

**Review by filipelautert — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,8 @@
+# Use the official Liquibase image as the base
+FROM liquibase/liquibase:4.29.1
+
+# Add the AWS License Service Extension using the Liquibase Package Manager (LPM)
+RUN lpm update && lpm add liquibase-aws-license-service
+
+# Default command to display Liquibase version
+CMD ["liquibase", "--version"]
```
Comment (StevenMassaro) [inline Dockerfile:8]: The main dockerfile calls the help command. Should we do the same here?

https://github.com/liquibase/docker/blob/main/Dockerfile#L60

File: README.md [hunk 3]
```diff
@@ -1,2 +1,39 @@
...
+
+   ![](./docs/image/logs_tab.png)
+
+8. To add more commands to test in the `aws-mp-test-cluster`, you can add them in the `Task Definitions` section.
+9. Contact the DevOps team to get access to the `LiquibaseAWSMP` AWS account or any other help required.
+
+# :blue_book: General Notes
+- This extension expects access to AWS, using standard AWS credentials mechanisms. Devs would need to get access to the following AWS account and set credentials from it: `LiquibaseAWSMP | 804611071420 | awsmp@liquibase.com`
+- This extension leverages a vulnerability in the `LicenseServiceFactory` in OSS/Pro. If that vulnerability is fixed, then this extension will no longer work. More info available [here](https://datical.atlassian.net/browse/DAT-12399).
+- This extension is released different than the other extensions because we need liquibase docker version to be released before deploying this extension to AWS Marketplace.
```
Comment (StevenMassaro) [inline README.md:39]: Please remove this section

```suggestion
```

File: .github/workflows/deploy-extension-to-marketplace.yml (hunk 1 above)
Comment (sayaliM0412) [reply .github/workflows/deploy-extension-to-marketplace.yml:20]: the below condition will take precedence over the `if:` 
`types:
      - closed
    paths:
      - 'Dockerfile'`

**Comment by StevenMassaro** [comment]
Just reminding you that this repo is public, and I don't know whether the stuff you put in the readme belongs in this repo or in a confluence page.

**Comment by sayaliM0412** [comment]
> Just reminding you that this repo is public, and I don't know whether the stuff you put in the readme belongs in this repo or in a confluence page.

good to have it here as well I think. Or I can create a link to confluence and add it to this README

**Comment by StevenMassaro** [comment]
> > Just reminding you that this repo is public, and I don't know whether the stuff you put in the readme belongs in this repo or in a confluence page.
> 
> good to have it here as well I think. Or I can create a link to confluence and add it to this README

As long as its okay being public, being in the repo is a better spot than in Confluence.

File: .github/workflows/deploy-extension-to-marketplace.yml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      steps:
+          - name: Checkout
+            uses: actions/checkout@v4
+
+          - name: Login to ECR Registry
+            uses: docker/login-action@v3
+            env:
+              AWS_REGION: us-east-1
+            with:
+              registry: 709825985650.dkr.ecr.us-east-1.amazonaws.com
```
Comment (jnewton03) [inline .github/workflows/deploy-extension-to-marketplace.yml:32]: should we make the registry a secret too?

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,8 @@
+# Use the official Liquibase image as the base
+FROM liquibase/liquibase:4.29.1
```
Comment (StevenMassaro) [inline Dockerfile:2]: So will this version change via dependabot whenever a new tag is released from the main docker image?

**Review by StevenMassaro — APPROVED** [review]

File: Dockerfile (hunk 5 above)
Comment (sayaliM0412) [reply Dockerfile:2]: yes

Comment (StevenMassaro) [reply Dockerfile:2]: Got it. So there will be a slight delay in publishing the AWS container version. That makes sense.

Comment (sayaliM0412) [reply Dockerfile:2]: yes, as the approval from official docker team takes sometime.

**Pull request merged**